[#118479861] Use new, password generating RDS broker

mtekel opened the pull request:
### What

[Generating secure master password for Postgres Instances](https://www.pivotaltracker.com/n/projects/1275640/stories/118479861)

Use latest broker from https://github.com/alphagov/paas-rds-broker/pull/11. Add new properties into spec and config file templates to configure the new features.
### Testing

Deploy this release against existing environment. E.g. by using https://github.com/alphagov/paas-cf/pull/320. Observe new broker release being deployed. In the broker logs you can see that it tries to connect first, fails ping and updates password afterwards. After deployment, all tests should pass and everything should continue working as usual.
### Merging

This PR depends on https://github.com/alphagov/paas-rds-broker/pull/11. Once that has been merged, remove https://github.com/alphagov/paas-aws-broker-boshrelease/commit/a3a6bfd117fb9e40ff5853f94df70c0dd91cdf3e and add a new commit pointing the rds-broker [submodule](https://github.com/alphagov/paas-aws-broker-boshrelease/tree/master/src/github.com/alphagov) at the merge commit of  https://github.com/alphagov/paas-rds-broker/pull/11.
### Who

not @mtekel or @keymon or @saliceti

keymon committed:
Add property rds-broker.master_password_seed

In order to generate secure unique keys per environment but keeping the
broker stateless, the broker will accept a initial password seed which
will be appended during the password generation.

The release operator must pass a secret string for this property.

This property is mandatory and has no default.

saliceti committed:
Add broker_name parameter to job config

It is used to tag DB instances in RDS and be able to retrieve them.

combor commented on jobs/rds-broker/spec:
```diff
@@ -38,6 +38,8 @@ properties:
   rds-broker.master_password_seed:
     description: "Secret seed to be used when generating the master RDS DB password"
+  rds-broker.broker_name:
+    description: "Unique name to identify by tag in RDS the instances created by this broker"
```
Ditto https://github.com/alphagov/paas-rds-broker/pull/11#discussion_r67344041

Commit:
Update submodule to the latest commit

We need to update submodule reference after merging
https://github.com/alphagov/paas-rds-broker/pull/11
